pyccel/codegen/printing/cwrappercode.py

```diff
@@ -75,15 +101,26 @@ def get_PyBuildValue(self, used_names, variable):
             collect_type = PyccelPyObject()
             collect_var = Variable(dtype=collect_type, rank = 1,
                 name = self.get_new_name(used_names, variable.name+"_tmp"))
-
             cast_function = self.get_cast_function(used_names, 'bool_to_pyobj', variable, collect_var)
-
+            return collect_var , cast_function
+        if variable.dtype is NativeComplex():
+            collect_type = PyccelPyObject()
+            collect_var = Variable(dtype=collect_type, rank = 1,
+                name = self.get_new_name(used_names, variable.name+"_tmp"))
+            cast_function = self.get_cast_function(used_names, 'complex_to_pycomplex', variable, collect_var)
             return collect_var , cast_function
         return variable, None
 
     def _print_PyccelPyObject(self, expr):
         return 'pyobject'
 
+    def _print_FuncCall(self, expr):
```

Comment:
Is this function necessary? Does FuncCall not print correctly if it inherits from FunctionCall ?

Reply:
Yes , Only if create a property getter  in the functioncall class that return  the name of the function  and use it directly in the printer .

Reply:
Is it not enough to save the name in the `_func` attribute?

That said, we probably can't use `FuncCall` without defining a `FunctionDef` as that will require access to the `FunctionDef` to get the types of the arguments/return values to ensure pointers are handled correctly